src/offline/cable_output.F90

```diff
@@ -1935,239 +1979,260 @@ SUBROUTINE write_output(dels, ktau, met, canopy, casaflux, casapool, casamet, ss
     CALL generate_out_write_acc(output%casa, ovid%LittCarbStructural, 'LittCarbStructural', out%LittCarbStructural, REAL(casapool%clitter(:, 2)/1000.0, 4) &
-                                , ranges%TotLittCarb, patchout%LittCarbStructural, met)
+                                , ranges%TotLittCarb, patchout%LittCarbStructural, out_settings)
     CALL generate_out_write_acc(output%casa, ovid%LittCarbCWD, 'LittCarbCWD', out%LittCarbCWD, REAL(casapool%clitter(:, 3)/1000.0, 4) &
-                                , ranges%TotLittCarb, patchout%LittCarbCWD, met)
+                                , ranges%TotLittCarb, patchout%LittCarbCWD, out_settings)
 
     ! cplant
     CALL generate_out_write_acc(output%casa, ovid%PlantCarbLeaf, 'PlantCarbLeaf', out%PlantCarbLeaf, REAL(casapool%cplant(:, 1)/1000.0, 4) &
-                                , ranges%TotLittCarb, patchout%PlantCarbLeaf, met)
+                                , ranges%TotLittCarb, patchout%PlantCarbLeaf, out_settings)
     CALL generate_out_write_acc(output%casa, ovid%PlantCarbWood, 'PlantCarbWood', out%PlantCarbWood, REAL(casapool%cplant(:, 2)/1000.0, 4) &
-                                , ranges%TotLittCarb, patchout%PlantCarbWood, met)
+                                , ranges%TotLittCarb, patchout%PlantCarbWood, out_settings)
     CALL generate_out_write_acc(output%casa, ovid%PlantCarbFineRoot, 'PlantCarbFineRoot', out%PlantCarbFineRoot, REAL(casapool%cplant(:, 3)/1000.0, 4) &
-                                , ranges%TotLittCarb, patchout%PlantCarbFineRoot, met)
+                                , ranges%TotLittCarb, patchout%PlantCarbFineRoot, out_settings)
 
     CALL generate_out_write_acc(output%casa, ovid%TotLivBiomass, 'TotLivBiomass', out%TotLivBiomass, REAL((SUM(casapool%cplant, 2)) &
-                                                                                                          /1000.0, 4), ranges%TotLivBiomass, patchout%TotLivBiomass, met)
+                                                                                                          /1000.0, 4), ranges%TotLivBiomass, patchout%TotLivBiomass, out_settings)
 
     IF (cable_user%sync_nc_file) &
       ok = NF90_SYNC(ncid_out)
 
   END SUBROUTINE write_output
 
-  SUBROUTINE acc_out_var_d1(output_var, out_var, acc_val)
-     LOGICAL, INTENT(IN) :: output_var ! Whether to write the current variable
-     REAL(4), DIMENSION(:) :: out_var
-     REAL(4), DIMENSION(:) :: acc_val
-
-     IF (output_var) THEN
-          ! Add current timestep's value to total of temporary output variable:
-          out_var = out_var + acc_val
-          IF (writenow) THEN
-               ! Divide accumulated variable by number of accumulated time steps:
-               out_var = out_var / REAL(output%interval, 4)
-          END IF
-     END IF
-  END SUBROUTINE acc_out_var_d1
+  PURE FUNCTION acc_out_var_d1(output_var, out_var, acc_val, writenow) RESULT(res)
```

Comment:
Actually, this function could be written and defined as an ELEMENTAL function. In which case, it can be applied to an array of any size. See: https://fortranwiki.org/fortran/show/elemental

This removes the need to have 2 functions and an interface.

Reply:
I see, initially I thought `elemental` required all arguments to be array, but the requirement is every argument should be scalar, and operations on any 2 arguments must conform to the same shape? (just want to clarify whether this is correct)

Reply:
Yes, you are correct. It's a bit different for functions and subroutines.

I found this great source of information: https://www.ibm.com/docs/en/xffbg/121.141?topic=procedures-elemental

Here is the excerpt for your question:

> The actual arguments in a reference to an elemental procedure can be either of the following:
> - All scalar. For elemental functions, if the arguments are all scalar, the result is scalar.
> - One or more array-valued. The following rules apply if one or more of the arguments is array-valued:
>    - For elemental functions, the shape of the result is the same as the shape of the array actual argument with the greatest rank. If more than one argument appears then all actual arguments must be conformable.
>    - For elemental subroutines, all actual arguments associated with [INTENT(OUT)](https://www.ibm.com/docs/en/SS2MB5_14.1.0/com.ibm.xlf141.bg.doc/language_ref/intent.html#intent) and INTENT(INOUT) dummy arguments must be arrays of the same shape, and the remaining actual arguments must be conformable with them.